src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+import { roundedDown } from './utility/roundedDown';
 
-export class Triangle implements Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => {};
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error("sides can't have zero length");
+    }
+
+    const longestSide = Math.max(a, b, c);
+    const perimeter = a + b + c;
+
+    if (longestSide >= perimeter - longestSide) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
```

Comment:
[CODE STYLE] - The error message should be more descriptive. It should explain that the sum of any two sides of a triangle must be greater than the third side.